### DISCUSSION TITLE
How can we read contract data even wallet not connected

### DISCUSSION THREAD

**Discussion by Rakeshpradhan1999** [wevm/wagmi#2829] [Q&A]
How can we read contract data even wallet not connected, for now we can read data while connected wallet but i want to read some specific data which is not related to wallet 

please let me know if any help available.

**Comment by tmm** [comment]
You can just use [`useContractRead`](https://wagmi.sh/react/hooks/useContractRead) or [`useContractReads`](https://wagmi.sh/react/hooks/useContractReads) without a wallet connection.

**Comment by abbasibneakram** [comment]
If you're read function dosn't require wallet address like in **balanceOf()** function then you can use useContractRead or useContractReads to rad the data because you must be using Alchemy or Infure node provider in the configuration of Wagmi , that helps us to interact with smart contract's read functions.
